client/state/partner-portal/licenses/handlers.ts

```diff
@@ -80,9 +80,12 @@ export function fetchLicensesHandler( action: HttpAction ): AnyAction {
 			path: '/jetpack-licensing/licenses',
 			query: {
 				// Do not apply filters during search as search takes over (matches Calypso Blue Post search behavior).
-				...( action.search ? { search: action.search } : { filter: action.filter } ),
+				...( action.search
+					? { search: action.search }
+					: { filter: action.filter, page: action.page } ),
```

Comment:
This means that search pagination will not work as it will never pass the `page` param. I think a better approach would be to reset the value of `page` in the search field `onSearch` callback (it may not be as easy due to how it's set up right now though) so whenever the search query changes the `page` param is reset to 1 or removed (currently `&page=X` remains in the window location even though it's not being passed to the API thus causing a desync between the URL and what we are showing to the user).

Reply:
I would suggest accepting this PR now if all the other things look good since the search currently only returns one result. And then I can address the improvement in another PR.